internal/hooks/paths_test.go

```
@@ -0,0 +1,37 @@
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSettingsPaths_GlobalAndProjects(t *testing.T) {
+	claude := t.TempDir()
+	// Global settings.
+	require.NoError(t, os.WriteFile(filepath.Join(claude, "settings.json"), []byte("{}"), 0o600))
+	// Two project settings.local.json files.
+	for _, proj := range []string{"projA", "projB"} {
+		dir := filepath.Join(claude, "projects", proj)
+		require.NoError(t, os.MkdirAll(dir, 0o755))
```

golangci-lint (2.12.2)

[medium] G301: Expect directory permissions to be 0750 or less

(gosec)